include/RevTracer.h

```diff
@@ -0,0 +1,229 @@
+//
+// _RevTracer_h_
+//
+// Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+// All Rights Reserved
+// contact@tactcomplabs.com
+//
+// See LICENSE in the top level directory for licensing details
+//
+//
+
+#ifndef _SST_REVCPU_REVTRACER_H_
+#define _SST_REVCPU_REVTRACER_H_
+
+// -- SST Headers
+#include "SST.h"
+
+// -- Standard Headers
+#include <cstdint>
+#include <ostream>
+#include <string>
+#include <iostream>
+#include <map>
+#include <vector>
+
+// -- Rev Headers
+
+// Integrated Disassembler (toolchain dependent)
+#ifdef REV_USE_SPIKE
+// sst and spike cause redefinition warning for macro UNUSED.
+// sst: #define UNUSED(x) x __attribute__((unused))
+// spike: # define UNUSED __attribute__ ((unused))
+// TODO can these be uniquified?
+#ifdef __GNUC__
+#undef UNUSED
+#endif
+#include "riscv/disasm.h"
+#endif
+
+// Tracing macros
+#ifndef NO_REV_TRACER
+#define TRACE_REG_READ(R,V)  { if (Tracer) Tracer->regRead(  (uint8_t) (R),(uint64_t) (V) ); }
+#define TRACE_REG_WRITE(R,V) { if (Tracer) Tracer->regWrite( (uint8_t) (R),(uint64_t) (V) ); }
+#define TRACE_PC_WRITE(PC)   { if (Tracer) Tracer->pcWrite( (uint64_t) (PC) ); }
+#define TRACE_MEM_WRITE(ADR, LEN, DATA) { if (Tracer) Tracer->memWrite( (ADR), (LEN), (DATA) ); }
+#define TRACE_MEM_READ(ADR, LEN, DATA)  { if (Tracer) Tracer->memRead(  (ADR), (LEN), (DATA) ); }
+#else
+#define TRACE_REG_READ(R,V)
+#define TRACE_REG_WRITE(R,V)
+#define TRACE_PC_WRITE(PC)
+#define TRACE_MEM_WRITE(ADR, LEN, DATA)
+#define TRACE_MEM_READ(ADR, LEN, DATA)
+#endif
+
+namespace SST{
+  namespace RevCPU{
```

Comment:
`namespace SST::RevCPU{` shorthand can be used instead of nested namespaces in C++17 and later.